### ISSUE TITLE
Trying to destroy a mounted file system causes daemon to become very confused

### ISSUE THREAD

**Issue by tasleson** [stratis-storage/stratisd#1137]
If we create a Stratis backed FS and mount it and then try to destroy it while it's mounted madness ensues...

```bash
# df -h
...
/dev/mapper/stratis-1-43957d9811384168ad682e2b0bf6cf0d-thin-fs-774d6b04f8834662848fa9291bb3143d  1.0T  1.1G 1023G   1% /mnt/destroy_while_mounted

# bin/stratis filesystem list
Pool Name  Name  Used      Created            Device                      
teststack  fs1   1.05 GiB  Aug 30 2018 10:54  /dev/stratis/teststack/fs1

# bin/stratis filesystem destroy teststack fs1
Execution failure caused by:
INTERNAL ERROR: low-level ioctl error

# bin/stratis filesystem list
Execution failure caused by:
no name for filesystem with uuid 774d6b04-f883-4662-848f-a9291bb3143d
    which in turn caused:
Error while invoking method "GetManagedObjects" belonging to interface "org.freedesktop.DBus.ObjectManager" with arguments ()

# bin/stratis pool list
Execution failure caused by:
no name for filesystem with uuid 774d6b04-f883-4662-848f-a9291bb3143d
    which in turn caused:
Error while invoking method "GetManagedObjects" belonging to interface "org.freedesktop.DBus.ObjectManager" with arguments ()

# bin/stratis blockdev list
Execution failure caused by:
no name for filesystem with uuid 774d6b04-f883-4662-848f-a9291bb3143d
    which in turn caused:
Error while invoking method "GetManagedObjects" belonging to interface "org.freedesktop.DBus.ObjectManager" with arguments ()

```

If we restart the daemon then all is well and if we un-mount the FS first we can destroy the FS.

```bash
# systemctl restart stratisd
[root@localhost stratis-cli]# bin/stratis filesystem list
Pool Name  Name  Used      Created            Device                      
teststack  fs1   1.05 GiB  Aug 30 2018 10:54  /dev/stratis/teststack/fs1
```

There is absolutely nothing to see in the journal when all this occurs :smiley:

**Comment by tasleson** [comment]
**Note:  The stratis pool is sitting on top of 3 encrypted NVME devices, but I don't think that matters**

**Comment by agrover** [comment]
oops yes, we should not allow mounted filesystems to be destroyed.

**Comment by agrover** [comment]
I looked at this a little. One complication is that `fs.destroy()` takes `self`, so if it fails you still no longer own the struct to put it back in the filesystem Table :frowning_face:. This also extends down into dm-rs's `ThinDev` code for destroy().

At this point I think we should probably revamp `destroy` methods in stratisd and dm-rs to take `&mut self`. (Taking `self` was kind of an experiment from the early days, and it seems clear now that `self` in + `Result` out = trouble.)

One result of this approach would be it'd be possible to have valid `ThinDev`s that no longer represent a mapped device. But for now this might be ok. You couldn't really do anything with it, since any method calls resulting in DM ioctls would fail. Stratisd's usage would be to immediately drop a successfully destroyed ThinDev, anyway.

**Comment by agrover** [comment]
should be fixed by #1159
